pallets/subtensor/src/macros/dispatches.rs

```diff
@@ -2152,6 +2153,115 @@ mod dispatches {
+        ) -> DispatchResult {
+            ensure!(Self::if_subnet_exist(netuid), Error::<T>::SubnetNotExists);
+            let maybe_owner = Self::ensure_subnet_owner_or_root(origin, netuid)?;
+
+            if schedule_set {
+                ensure!(
+                    SubnetDeregistrationPrioritySchedule::<T>::get(netuid).is_none(),
+                    Error::<T>::SubnetDeregistrationPriorityAlreadyScheduled
+                );
+
+                let current_block: BlockNumberFor<T> = <frame_system::Pallet<T>>::block_number();
+                let delay: BlockNumberFor<T> = ColdkeySwapScheduleDuration::<T>::get();
+                let when: BlockNumberFor<T> = current_block.saturating_add(delay);
+
+                let call = Call::<T>::force_set_deregistration_priority { netuid };
+                let bound_call = <T as Config>::Preimages::bound(LocalCallOf::<T>::from(call))
+                    .map_err(|_| Error::<T>::FailedToSchedule)?;
+
+                T::Scheduler::schedule(
+                    DispatchTime::At(when),
+                    None,
+                    63,
+                    frame_system::RawOrigin::Root.into(),
+                    bound_call,
+                )
+                .map_err(|_| Error::<T>::FailedToSchedule)?;
+
+                SubnetDeregistrationPrioritySchedule::<T>::insert(netuid, when);
+
+                Self::deposit_event(Event::SubnetDeregistrationPriorityScheduled(netuid, when));
+            } else {
+                if maybe_owner.is_some() {
+                    if SubnetDeregistrationPrioritySchedule::<T>::take(netuid).is_some() {
+                        Self::deposit_event(Event::SubnetDeregistrationPriorityScheduleCleared(
+                            netuid,
+                        ));
+                    }
+                } else {
+                    let was_queued = Self::remove_subnet_from_deregistration_priority_queue(netuid);
+                    let had_schedule =
+                        SubnetDeregistrationPrioritySchedule::<T>::take(netuid).is_some();
+
+                    if was_queued {
+                        Self::deposit_event(Event::SubnetDeregistrationPriorityCleared(netuid));
+                    }
+                    if had_schedule {
+                        Self::deposit_event(Event::SubnetDeregistrationPriorityScheduleCleared(
+                            netuid,
+                        ));
+                    }
+                }
+            }
+
+            Ok(())
+        }
+
+        /// Enqueues the subnet for deregistration immediately.
+        ///
+        /// This call is intended to be used by the scheduler and requires root origin.
+        #[pallet::call_index(122)]
+        #[pallet::weight((
+            Weight::from_parts(15_000_000, 0)
+                .saturating_add(T::DbWeight::get().reads_writes(3, 2)),
+            DispatchClass::Operational,
+            Pays::No
+        ))]
+        pub fn force_set_deregistration_priority(
```

Comment:
```suggestion
        pub fn enqueue_subnet_deregistration(
```

I'd definitely go with naming similar to what we have for coldkey swap. Plus, `force_` might be misleading when someone new reads the code. Extrinsics imo should be as descriptive as possible hence I'd go with "subnet deregistration" rather than "deregistration priority" as the term "deregistration" is not reserved solely for subnets but also for miners/validators.